### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by chamons**
As autorest.csharp gains increased traffic with more developers, laying out expectations for review and submission with a PR template (like azure-rest-api-specs does) seemed reasonable.

Consider the text draft and not set in stone - I just wrote up the most common PR mistakes I've seen in the past. 

Thoughts?

### CODE REVIEW

**Review by ShivangiReja — APPROVED** [review]

**Comment by chamons** [comment]
Thoughts @m-nash? Currently blocking until I get a 👍  / 👎 .

**Comment by allenjzhang** [comment]
Thanks, Chris for capturing these best practices. As we discussed in the meeting:

- We will add additional CODEOWNER for the mgmt path and common so that right folks get tagged for review and approval
- Reasonable and sufficient time is given to reviewers on PRs particularly for complex ones. 
- Draft PRs are not actively reviewed unless explicitly @someone to get an early eyeball on them.
- These are the general good principal/practices to follow. Though in some cases, due to blockage and/or time sensitivities, issues/discussions can be deferred post Merge.
- Educate folks on git workflow on fetch/cherry pick items in PR to reduce sequential blockage and afford more time on review as much as possible.

**Review by allenjzhang — APPROVED** [review]

**Pull request merged**
